sources/packages/backend/apps/api/src/services/application-bulk-withdrawal/application-bulk-withdrawal-import-business-validation.service.ts

```diff
@@ -0,0 +1,149 @@
+   * combined with the silently parameter defined as false.
+   */
+  validateApplicationBulkWithdrawalBusinessModels(
+    businessValidationModels: ApplicationBulkWithdrawalImportBusinessValidationModel[],
+    silently = false,
+  ): ApplicationWithdrawalValidationResult[] {
+    return businessValidationModels.map((model, index) => {
+      // Ensures that the object received is a class. This is needed to the
+      // proper validation metadata be available to the validation be performed.
+      const applicationBulkWithdrawalBusinessValidationModel = plainToClass(
+        ApplicationBulkWithdrawalImportBusinessValidationModel,
+        model,
+        {
+          enableImplicitConversion: true,
+        },
+      );
+      const validationsErrors = validateSync(
+        applicationBulkWithdrawalBusinessValidationModel,
+      );
+      const allErrors: string[] = [];
+      const allWarnings: ValidationResult[] = [];
+      validationsErrors.forEach((error) => {
+        const validation = this.extractValidationsByType(error);
+        allErrors.push(...validation.errors);
+        allWarnings.push(...validation.warnings);
+      });
+
+      if (!silently && !!allErrors.length) {
+        throw new CustomNamedError(
+          "The application bulk withdrawal has critical errors.",
+          APPLICATION_BULK_WITHDRAWAL_VALIDATION_CRITICAL_ERROR,
+          allErrors,
+        );
+      }
+
+      return {
+        index,
+        applicationBulkWithdrawalBusinessValidationModel,
+        warnings: allWarnings,
+        errors: allErrors,
+      };
+    });
+  }
+
+  /**
+   * Verify if all application withdrawal business validations were performed with success and throw
+   * a BadRequestException in case of some failure.
+   * @param businessValidationsResult business validation results to be verified.
+   */
+  assertBusinessValidationsAreValid(
+    businessValidationsResult: ApplicationWithdrawalValidationResult[],
+  ) {
+    const businessValidations = businessValidationsResult.filter(
+      (businessValidationResult) =>
+        businessValidationResult.errors.length ||
+        businessValidationResult.warnings.length,
+    );
+    if (businessValidations.length) {
+      // At least one error or warning was detected.
+      const errorValidationResults: ApplicationBulkWithdrawalValidationResultAPIOutDTO[] =
+        businessValidations.map((validation) => ({
+          recordIndex: validation.index,
+          applicationNumber:
+            validation.applicationBulkWithdrawalBusinessValidationModel
+              .applicationNumber,
+          withdrawalDate:
+            validation.applicationBulkWithdrawalBusinessValidationModel
+              .withdrawalDate,
+          sin: validation.applicationBulkWithdrawalBusinessValidationModel.sin,
+          errors: validation.errors,
+          warnings: validation.warnings,
+          infos: [],
+        }));
+      throw new BadRequestException(
```

Comment:
We do not throw HTTP errors from the service layer, only from `controller services`.